File: Model/src/main/java/model/domain/Reservation.java
package model.domain;

import model.utils.Pair;

public class Reservation implements Identifiable<Pair<String, Trip>> {
    private String client;
    private Trip trip;
    private final String agency;
    private final String phoneNumber;
    private final int seats;

    public Reservation(Pair<String, Trip> reservationID, String agency, String phoneNumber, int seats) {
        this.client = reservationID.getFirst();
        this.trip = reservationID.getSecond();
        this.agency = agency;
        this.phoneNumber = phoneNumber;
        this.seats = seats;
    }

    @Override
    public Pair<String, Trip> getId() {
        return new Pair<>(client, trip);
    }

    @Override
    public void setId(Pair<String, Trip> id) {
        client = id.getFirst();
        trip = id.getSecond();
    }

    public String getClient() {
        return client;
    }

    public Trip getTrip() {
        return trip;
    }

    public String getPhoneNumber() {
        return phoneNumber;
    }

    public int getSeats() {
        return seats;
    }

    public String getAgency() {
        return agency;
    }

    @Override
    public String toString() {
        return "Reservation by " + this.client + " on trip to " + this.trip.getTouristAttraction()
                + " by " + this.agency + " for " + this.seats + " seats";
    }
}

File: Model/Reservation.cs
using System;

namespace Model
{
    [Serializable]
    public class Reservation : IIdentifiable<Tuple<string, Trip>>
    {
        public string Client { get; set;  }
        public Trip Trip { get; set; }
        public string Agency { get; set; }
        public string PhoneNumber { get; set; }
        public int Seats { get; set; }

        public Reservation(Tuple<string, Trip> id, string agency, string phoneNumber, int seats)
        {
            Client = id.Item1;
            Trip = id.Item2;
            Agency = agency;
            PhoneNumber = phoneNumber;
            Seats = seats;
        }

        public Tuple<string, Trip> getId()
        {
            return new Tuple<string, Trip>(Client, Trip);
        }

        public void setId(Tuple<string, Trip> id)
        {
            Client = id.Item1;
            Trip = id.Item2;
        }

        public override string ToString()
        {
            return "Reservation by " + Client + " on trip to " + Trip.TouristAttraction + " for " + Seats + " seats";
        }
    }
}
